### ISSUE TITLE
error while trying to download lexicon free AM model

### ISSUE THREAD

**Issue by cri5Castro** [facebookresearch/wav2letter#523]
hi there i'm trying to download the  pretrained librispeech acoustic model from  [lexiconFree](https://github.com/facebookresearch/wav2letter/tree/9bf45382f0d9c8707019f3950634c9ffad385272/recipes/models/lexicon_free/librispeech) recipe but there is a forbidden error , also i was able to download it before but i got 
```
E0206 23:18:17.085983   950 Serial.h:74] Error while loading: Failed to read 4 bytes from input stream! Read 0 
```
while trying to fork the model

**Comment by tlikhomanenko** [comment]
Hi @cri5Castro,

you can download the model from here https://github.com/facebookresearch/wav2letter/tree/master/recipes/models/lexicon_free/librispeech (from the current master, the link is fixed, there was problem with '+' in the name). Direct link https://dl.fbaipublicfiles.com/wav2letter/lexicon_free/librispeech/models/am/baseline_dev-clean%2Bother.bin.

Can you try to continue training from this model?

**Comment by cri5Castro** [comment]
thanks for the reply I can download it now but i'm getting an error while trying to fork the model
```
E0210 23:21:11.821619    43 Serial.h:74] Error while loading: Failed to read 4 bytes from input stream! Read 0
```
i can only train from scratch for some reason (my tokens list are diferent than the pretrained model) I'm training for spanish (char level)  trying to take advantace of the pretrained weights as starting point

**Comment by tlikhomanenko** [comment]
Is the size of your token set different from the one I used in lexicon free recipe? If you set another token set and lexicon and token set size is different from the one I used then it should fail because dimension of the last layer is different. You need to do some coding: loading model, cut the last layer, add new layer with your output dim and then saving it. After you can use it to train on your set.

**Comment by cri5Castro** [comment]
thanks yep I thought that so, thanks for the reply :)

**Comment by cri5Castro** [comment]
update: I was able to perfom the fork (finetuning) using the code from #507 provided by @vineelpratap , don't forget to set the --datadir flag since i believe another one is stored in the pretrained bin

**Comment by tlikhomanenko** [comment]
Yep, you should be careful to set correct `--datadir` and `--tokensdir`, they are using as prefixes and also during fork or continue all flags are loaded from the original model.

Closing the issue. Feel free to reopen!
